crates/storage/db-models/src/client_version.rs

```diff
@@ -29,9 +29,10 @@ impl reth_codecs::Compact for ClientVersion {
     where
         B: bytes::BufMut + AsMut<[u8]>,
     {
-        self.version.to_compact(buf);
-        self.git_sha.to_compact(buf);
-        self.build_timestamp.to_compact(buf)
+        let version_size = self.version.to_compact(buf);
+        let git_sha_size = self.git_sha.to_compact(buf);
+        let build_timestamp_size = self.build_timestamp.to_compact(buf);
+        version_size + git_sha_size + build_timestamp_size
```

Comment:
I believe this is correct and is not problematic because the string decode fns don't even require a len param

https://github.com/paradigmxyz/reth/blob/641f99ffdaa13c6be04474431c0d2d8d9e5d70e2/crates/storage/codecs/src/lib.rs#L194-L194